src/behavior_uct_single_agent.hpp

```diff
@@ -48,8 +48,8 @@ class BehaviorUCTSingleAgent : public BehaviorModel {
     double return_upper_bound_;
 };
 
-inline BehaviorModel *BehaviorUCTSingleAgent::Clone() const {
-  return new BehaviorUCTSingleAgent(*this);
+inline std::shared_ptr<modules::models::behavior::BehaviorModel> BehaviorUCTSingleAgent::Clone() const {
```

Comment:
as we want to return a Behavior Model Ptr, I think we should 
use dynamic_pointer_cast,
also omit modules::models:behavior 

`inline std::shared_ptr<BehaviorModel> BehaviorUCTSingleAgent::Clone() const {`
  `std::shared_ptr<BehaviorUCTSingleAgent> model_ptr = std::make_shared<BehaviorUCTSingleAgent>(*this);`
  `return std::dynamic_pointer_cast<BehaviorModel>(model_ptr);`
`}`